src/OpenID/Connect/Client/Flow/AuthorizationCode.hs

```diff
@@ -412,9 +412,13 @@ exchangeCodeForIdentityToken https now disco creds user = do
       & bimap InvalidProviderTokenResponseError fst
       >>= (decodeIdentityToken >>> first TokenDecodingError)
 
-    authMethods :: [ClientAuthentication]
-    authMethods = maybe [ClientSecretPost] NonEmpty.toList
-      (tokenEndpointAuthMethodsSupported disco)
+    stdAuthMethods :: [ClientAuthentication]
```

Comment:
Related to the comment above, I don't like how any authentication extensions are silently stripped off here.  I think we need to force the library user to do this elsewhere.